docs/third-party-developers/extensibility/rest-api/extend-rest-api-add-custom-fields.md

```diff
@@ -0,0 +1,176 @@
+
+```json
+"parent": [ "woocommerce/checkout-shipping-methods-block" ],
+"attributes": {
+	"lock": {
+		"type": "object",
+		"default": {
+			"remove": true,
+			"move": true
+		}
+	}
+}
+```
+
+- The [lock attribute](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-templates/#individual-block-locking) is an object that controls whether the block can be removed or moved. By default, the lock attribute is set to allow the block to be removed and moved. However, by modifying the lock attribute, you can “force” the block to be non-removable. For example, you can set both remove and move properties to false in order to prevent the block from being removed or moved.
+- The [parent attribute](https://developer.wordpress.org/block-editor/reference-guides/block-api/block-metadata/#parent) specifies the parent block that this block should be nested within it. It determines where the block will render. In our example, the block is a child of the `woocommerce/checkout-shipping-methods-block`. This means that the your block will be rendered within the `woocommerce/checkout-shipping-methods-block`.
+
+### 3. Setting custom checkout data
+
+We can set the added field data to send it to the `wc/store/checkout` endpoint when processing orders using the function `setExtensionData`:
+
+```JavaScript
+setExtensionData(
+	'namespace-of-your-block',
+	'key-of-your-data',
+	value
+);
+```
+
+#### Parameters
+
+- namespace `string` - The namespace of your block.
+- key `string` - The key of your data.
+- value `any` - The value of your data.
+
+### 4. Processing the Checkout POST Request
+
+To process the added field data, extend the Store API. We can do this within the `initialize` of your `IntegrationInterface` class.
+
+#### Code Example
+
+We will use the following PHP files in our example:
+
+- The `custom-inner-block-blocks-integration.php` file:
+
+```php
+use Automattic\WooCommerce\Blocks\Integrations\IntegrationInterface;
+
+/**
+ * Class for integrating with WooCommerce Blocks
+ */
+class Custom_Inner_Block_Blocks_Integration implements IntegrationInterface {
+	// ... Some code here
+
+	/**
+	 * When called invokes any initialization/setup for the integration.
+	 */
+	public function initialize() {
+		require_once __DIR__ . '/custom-inner-block-extend-store-endpoint.php';
+		// ... Some code here: (e.g. init functions that registers scripts and styles, and other instructions)
+		$this->extend_store_api();
+	}
+
+	/**
+	 * Extends the cart schema to include the custom-inner-block value.
+	 */
+	private function extend_store_api() {
+		Custom_Inner_Block_Extend_Store_Endpoint::init();
+	}
+}
+```
+
+- The `custom-inner-block-extend-store-endpoint.php` file: extends the [Store API](https://github.com/woocommerce/woocommerce-blocks/tree/trunk/src/StoreApi) and adds hooks to save and display your new field block instructions.
```

Comment:
This doesn't save the data from the custom block anywhere by default, the developer would have to be explicit about this.